rbc/heavydb/geo_nested_array.py

```diff
@@ -0,0 +1,336 @@
+def heavydb_geo_isNull(geo):
+    if isinstance(geo, GeoNestedArrayNumbaType):
+        base_type = geo.base_type
+        isNull = external(f"bool {base_type}_isNull(int8_t*)|cpu")
+
+        def impl(geo):
+            return isNull(as_voidptr(get_alloca(geo)))
+
+        return impl
+
+
+@extending.overload(operator.getitem)
+@extending.overload_method(GeoNestedArrayNumbaType, "get_item")
+def heavydb_geo_getitem(geo, index):
+    if isinstance(geo, GeoNestedArrayNumbaType):
+
+        def impl(geo, index):
+            return heavydb_geo_getitem_(geo, index)
+
+        return impl
+
+
+def heavydb_geo_fromCoords_vec(geo, lst):
+    base_type = geo.base_type
+    fromCoords = external(
+        f"void {base_type}_fromCoords(int8_t*, double*, int64_t size)|cpu"
+    )
+
+    def impl(geo, lst):
+        sz = len(lst)
+
+        f64p = get_c_ptr(sz, nb_types.double)
+        for i in range(sz):
+            f64p[i] = lst[i]
+
+        return fromCoords(as_voidptr(get_alloca(geo)), f64p, sz)
+
+    return impl
+
+
+def heavydb_geo_fromCoords_vec2(geo, lst):
+    base_type = geo.base_type
+    fromCoords = external(
+        f"void {base_type}_fromCoords(int8_t*, double*, int64_t*, int64_t)|cpu"
+    )
+
+    def impl(geo, lst):
+        nrows = len(lst)
+        geo_ptr = as_voidptr(get_alloca(geo))
+
+        indices = get_c_ptr(nrows, nb_types.int64)
+        n_elems = 0
+        for i in range(nrows):
+            n_elems += len(lst[i])
+            indices[i] = len(lst[i])
+
+        data = get_c_ptr(n_elems, nb_types.double)
+        idx = 0
+        for i in range(nrows):
+            ncols = len(lst[i])
+            for j in range(ncols):
+                data[idx] = lst[i][j]
+                idx += 1
+
+        return fromCoords(geo_ptr, data, indices, nrows)
+
+    return impl
+
+
+def heavydb_geo_fromCoords_vec3(geo, lst):
+    base_type = geo.base_type
+    fromCoords = external(
+        f"void {base_type}_fromCoords(int8_t*, double*, int64_t*, int64_t*, int64_t, int64_t)|cpu"
+    )
+
+    def impl(geo, lst):
+        geo_ptr = as_voidptr(get_alloca(geo))
+
+        indices_i = []
+        indices_j = []
+        flatten = []
+
+        n_matrices = len(lst)
+        for i in range(n_matrices):
+            cnt = 0
+            nrows = len(lst[i])
+            for j in range(nrows):
+                ncols = len(lst[i][j])
+                cnt += ncols
+                indices_j.append(ncols)
+                for k in range(ncols):
+                    flatten.append(lst[i][j][k])
+            indices_i.append(cnt)
+
+        data = get_c_ptr(len(flatten), nb_types.double)
+        for i, e in enumerate(flatten):
+            data[i] = e
+
+        indices_i_ = get_c_ptr(len(indices_i), nb_types.int64)
+        for i, e in enumerate(indices_i):
+            indices_i_[i] = e
+
+        indices_j_ = get_c_ptr(len(indices_j), nb_types.int64)
+        for j, e in enumerate(indices_j):
+            indices_j_[j] = e
```

Comment:
General note: these loops are run in Python. Is this true or is the `impl` function jitted with numba?
If true, this implies that using `fromCoords` will not be the most efficient way to construct Geo data instances from Python.

Reply:
The function is jitted with Numba